src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -526,13 +527,8 @@ protected override AuthenticationTicket AuthenticateCore()
         /// <param name="nonce">the nonce to remember.</param>
         /// <remarks><see cref="HttpResponse.Cookies.Append"/>is called to add a cookie with the name: 'OpenIdConnectAuthenticationDefaults.Nonce + <see cref="OpenIdConnectAuthenticationOptions.StringDataFormat.Protect"/>(nonce)'.
         /// The value of the cookie is: "N".</remarks>
-        private void WriteNonceCookie(string nonce)
+        private void WriteNonceCookie([NotNull] string nonce)
```

Comment:
NotNull doesn't quite replace IsNullOrEmpty. (Though we've been told to avoid IsNullOrWhitespace).